client/components/email-verification/index.js

```diff
@@ -12,6 +13,20 @@ export default function emailVerification( context, next ) {
 	const showVerifiedNotice = '1' === context.query.verified;
 	const showNewEmailNotice = '1' === context.query.new_email_result;
 
+	// Redirect email verification to v2 for users who have opted into the new Hosting Dashboard
+	if (
+		showNewEmailNotice &&
+		( context.pathname === '/me/account' || context.pathname === '/settings/account' )
+	) {
+		const state = context.store.getState();
+		const hostingDashboardPreference = getPreference( state, 'hosting-dashboard-opt-in' );
+
+		if ( hostingDashboardPreference === 'opt-in' ) {
```

Comment:
```suggestion
		if ( hostingDashboardPreference.value === 'opt-in' ) {
```
Here's what's missing to make it work ;)

Reply:
Oops, looks like this fell off as I was testing things 🙈 Used the `hasHostingDashboardOptIn` helper in [302dded](https://github.com/Automattic/wp-calypso/pull/106103/commits/302dded34dd9cb6c774ab8278932fea347001fdc)!